src/App.tsx

```diff
@@ -1,26 +1,189 @@
-/* eslint-disable max-len */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/dot-notation */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { USER_ID } from './api/todos';
+import classNames from 'classnames';
+import { Todo } from './types/Todo';
+import * as todosFromServer from './api/todos';
+import { wait } from './utils/fetchClient';
+import { TodoForm } from './components/TodoForm/TodoForm';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoFooter } from './components/TodoFooter/TodoFooter';
+import { Status } from './types/Status';
+import { TodoItem } from './components/TodoItem/TodoItem';
 
-const USER_ID = 0;
+const getTodosByStatus = (status: string, todos: Todo[]) => {
+  const preperedTodos = [...todos];
+
+  if (status) {
+    switch (status) {
+      case Object.keys(Status)[Object.values(Status).indexOf(Status.active)]:
+        return preperedTodos.filter(todo => !todo.completed);
+      case Object.keys(Status)[Object.values(Status).indexOf(Status.completed)]:
+        return preperedTodos.filter(todo => todo.completed);
+      default:
+        return preperedTodos;
+    }
+  }
+
+  return preperedTodos;
+};
 
 export const App: React.FC = () => {
+  const [titleError, setTitleError] = useState(false);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loadError, setLoadError] = useState(false);
+  const [addError, setAddError] = useState(false);
+  const [deleteError, setDeleteError] = useState(false);
+  const [status, setStatus] = useState('all');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const filteredTodos = getTodosByStatus(status, todos);
+
+  function addTodo(newTodoTitle: string): any {
+    const editedTitle = newTodoTitle.trim();
+
+    if (!editedTitle) {
+      setTitleError(true);
+      wait(3000).then(() => setTitleError(false));
+
+      return;
+    } else {
+      setTempTodo({
+        id: 0,
+        userId: 839,
+        title: editedTitle,
+        completed: false,
+      });
+
+      return todosFromServer
+        .createTodos({
+          userId: 839,
+          title: editedTitle,
+          completed: false,
+        })
+        .then(newTodo => {
+          setTodos(prevTodos => [...prevTodos, newTodo]);
+          setTempTodo(null);
+        })
+        .catch(error => {
+          setAddError(true);
+          setTempTodo(null);
+          wait(3000).then(() => setAddError(false));
+          throw error;
+        });
+    }
+  }
+
+  function updateTodo(updatedTodo: Todo) {
```

Comment:
It is important to follow one way of creating functions
```suggestion
  const updateTodo = (updatedTodo: Todo) => {
```